App.tsx

```diff
@@ -134,10 +134,10 @@ const BaseApp: React.FunctionComponent<{
   });
 
   const onLayoutRootView = useCallback(async () => {
-    if (fontsLoaded) {
-      await SplashScreen.hideAsync();
-    }
-  }, [fontsLoaded]);
+    // This callback won't execute until fontsLoaded is true, because
+    // otherwise we won't render the view that triggers this callback
+    await SplashScreen.hideAsync();
+  }, []);
```

Comment:
didn't need to re-evaluate `useCallback` because it would never be executed in the `fontsLoaded === false` case